File: easy/1.two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        h = {num:i for i, num in enumerate(nums)}
        for i, num in enumerate(nums):
            need = target - num
            if need not in h:
                continue
            if i == h[need]:
                continue
            return [i, h[need]]
        

File: easy/1.two-sum.go
package main
func twoSum(nums []int, target int) []int {
    l := len(nums)
    for i := 0; i < l-1; i++ {
        target2 := target - nums[i]
        for j := i+1; j < l; j++ {
            //fmt.Println(i,j, target2)
            if  target2 == nums[j] {
                return []int{i,j}
            }
        }
    }
    return nil
}